### PR TITLE
not all disk types can be selected as the boot source

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

**Comment by glekner** [comment]
image

@matthewcarleton @jelkosz can you approve the validation message

**Comment by glekner** [comment]
@suomiy can you review

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/storage-tab/storage-boot-source.tsx [hunk 1]
```diff
@@ -56,11 +65,9 @@ export const StorageBootSource: React.FC<StorageBootOrderProps> = ({
           isDisabled={isDisabled}
         >
           <FormSelectPlaceholderOption isDisabled placeholder={SELECT_BOOTABLE_DISK} />
-          {ignoreCaseSort(filteredStorages, null, (storage) => storage.disk?.name).map(
-            (storage) => (
-              <FormSelectOption key={storage.id} value={storage.id} label={storage.disk?.name} />
-            ),
-          )}
+          {ignoreCaseSort(storages, null, (storage) => storage.disk?.name).map((storage) => (
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/storage-tab/storage-boot-source.tsx:68]: why are we including the disks they cannot select?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
 export const ATTACH_CD = 'Attach CD-ROM';
-export const NO_BOOTABLE_ATTACHED_DISK_ERROR = 'A bootable attached disk could not be found';
+export const BOOTABLE_ATTACHED_DISK_MESSAGE = 'Boot disks can not have a blank source';
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts:2]: blank source has a special meaning. Shouldn't we say something like that `selecting a boot source is required`?

**Review by suomiy — CHANGES_REQUESTED** [review]

**Comment by glekner** [comment]
added `CONTAINER_DISK` as a viable boot source @suomiy

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/storage-tab/storage-boot-source.tsx (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/storage-tab/storage-boot-source.tsx:69]: ok let's show all disks

**Comment by suomiy** [comment]
+1, just this left https://github.com/openshift/console/pull/7080#discussion_r516660247

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts (hunk 2 above)
Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts:2]: @matthewcarleton can you jump in?

Comment (matthewcarleton) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts:2]: What we want to communicate is that the disk they are selecting must have a source defined and can not be blank. 
Could we say:
"A boot source disk must have a source and can not be blank"

**Comment by glekner** [comment]
@suomiy updated string

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts (hunk 2 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/strings/storage.ts]: not sure if we want to mention other invalid sources (which is not just blank type), but ok let's put this message

**Pull request merged**